### PR TITLE
i18n(de): translate guides/site-search

### PR DESCRIPTION
**Pull request by trueberryless**
#### Description

Translation of `guides/site-search.mdx` into German language.

### CODE REVIEW

**Comment by trueberryless** [comment]
Links fixed in #2476 and #2459

File: docs/src/content/docs/de/guides/site-search.mdx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+   	docs: defineCollection({ schema: docsSchema() }),
+   	i18n: defineCollection({
+   		type: 'data',
+   		schema: i18nSchema({ extend: docSearchI18nSchema() }),
+   	}),
+   };
+   ```
+
+2. Füge Übersetzungen zu deinen JSON-Dateien in `src/content/i18n/` hinzu.
+   Hier sind die englischen Standardwerte, die von DocSearch verwendet werden:
```
Comment (HiDeoo) [inline docs/src/content/docs/de/guides/site-search.mdx:136]:
```suggestion

   Hier sind die englischen Standardwerte, die von DocSearch verwendet werden:
```

Nit: extra new line to match the English version.

**Review by HiDeoo — COMMENTED** [review]
Thanks for the translation :raised_hands:

As a non-native speaker, I reviewed this PR using the [Docs Translation Review Checklist](https://contribute.docs.astro.build/reviewers/reviewing-translations/) and everything looks good to me 👍 Once links validation passes, this PR should mergeable.

**Comment by trueberryless** [comment]
Thanks for the review, should be good to go!

File: docs/src/content/docs/de/guides/site-search.mdx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+   ```
+
+</Steps>
+
+Mit dieser aktualisierten Konfiguration öffnet die Suchleiste deiner Website nun ein Algolia-Modal anstelle des Standardmodals.
+
+#### Die Benutzeroberfläche von DocSearch übersetzen
+
+DocSearch stellt standardmäßig nur englische Benutzeroberflächen-Strings zur Verfügung.
+Füge mithilfe des in Starlight integrierten [Internationalisierungssystems](/de/guides/i18n/#starlights-ui-%C3%BCbersetzen) Übersetzungen der Benutzeroberfläche der Modale für deine Sprache hinzu.
```
Comment (HiDeoo) [inline docs/src/content/docs/de/guides/site-search.mdx:114]:
```suggestion
Füge mithilfe des in Starlight integrierten [Internationalisierungssystems](/de/guides/i18n/#starlights-ui-übersetzen) Übersetzungen der Benutzeroberfläche der Modale für deine Sprache hinzu.
```

This should fix the last remaining link issue.

**Review by HiDeoo — APPROVED** [review]
Everything looks great now, ready to be merged, thanks for the translation :rocket:

**Pull request merged**